cornice/schemas.py

```diff
@@ -175,3 +175,78 @@ def _validate_fields(location, data):
         for param in set(params) - set([attr.name for attr in attrs]):
             request.errors.add('body' if param in body else 'querystring',
                                param, msg % param)
+
+
+def validate_matchdict(matchdict_types):
```

Comment:
The solution you have posted may be alright if you want to use Colander, however if you want to use other libraries for schema validation, this may not be the case. I think for some use cases my solution is much simpler and more performant.